From 63c8189910c1cc2a45a7fb21fbc5c516ce01f46c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jakub=20=C4=8Cajka?= <jcajka@redhat.com>
Date: Thu, 22 Mar 2018 12:41:12 +0100
Subject: [PATCH 2/6] Don't use the bundled tzdata

---
 src/time/zoneinfo_unix.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/time/zoneinfo_unix.go b/src/time/zoneinfo_unix.go
index bbf263a16f..ec14bc0f7b 100644
--- a/src/time/zoneinfo_unix.go
+++ b/src/time/zoneinfo_unix.go
@@ -32,14 +32,14 @@ var zoneDirs = []string{
 	"/usr/share/zoneinfo/",
 	"/usr/share/lib/zoneinfo/",
 	"/usr/lib/locale/TZ/",
-	runtime.GOROOT() + "/lib/time/zoneinfo.zip",
 }
 
 var origZoneDirs = zoneDirs
 
 func forceZipFileForTesting(zipOnly bool) {
-	zoneDirs = make([]string, len(origZoneDirs))
+	zoneDirs = make([]string, len(origZoneDirs)+1)
 	copy(zoneDirs, origZoneDirs)
+	zoneDirs = append(zoneDirs, runtime.GOROOT()+"/lib/time/zoneinfo.zip")
 	if zipOnly {
 		for i := 0; i < len(zoneDirs)-1; i++ {
 			zoneDirs[i] = "/XXXNOEXIST"
-- 
2.14.3

