Index: go/src/pkg/os/os_test.go
===================================================================
--- go.orig/src/pkg/os/os_test.go
+++ go/src/pkg/os/os_test.go
@@ -842,7 +842,7 @@ func run(t *testing.T, cmd []string) str
 	if err != nil {
 		t.Fatal(err)
 	}
-	p, err := StartProcess("/bin/hostname", []string{"hostname"}, &ProcAttr{Files: []*File{nil, w, Stderr}})
+	p, err := StartProcess("/usr/bin/hostname", []string{"hostname"}, &ProcAttr{Files: []*File{nil, w, Stderr}})
 	if err != nil {
 		t.Fatal(err)
 	}
